tests/pipelines/stepping.py

```diff
@@ -0,0 +1,148 @@
+"""
+Pipeline stepping is a way to advance the asynchronous data pipeline
+deterministically by exposing each step separately and advancing the states
+manually.
+"""
+
+
+import functools
+import time
+import types
+import logging
+from contextlib import contextmanager
+from unittest.mock import patch
+
+import bigchaindb.core
+from bigchaindb.backend.changefeed import ChangeFeed
+import bigchaindb.pipelines.block
+import bigchaindb.pipelines.stale
+import bigchaindb.pipelines.vote
+
+
+class MultipipesStepper:
```

Comment:
I like this as well (I was trying to achieve something similar inside Multipipes, it was called [*sequential strategy*](https://github.com/bigchaindb/multipipes/blob/master/tests/test_sequential_strategy.py) but it was considering the pipeline as a whole). Would be nice to have some of this code in the `multipipes` module.

The only thing I don't like that much is the signature of the method, that is `stepper.<pipeline_name>_<pipeline_function>`, why not having `stepper.<pipeline_name>.<pipeline_function>`?